src/config/network/iwd.md

```diff
@@ -13,59 +13,62 @@ Install the `iwd` package and enable the `dbus` and `iwd` services.
-[iwd]# station <interface> disconnect                    # Disconnect from a network
-[iwd]# known-networks list                               # List known networks
-[iwd]# known-networks forget <network-name>              # Forget a known network
-```
-
-> Note: `iwctl` also supports passing commands as arguments.
-> 
-> ```
-> $ iwctl station <interface> get-networks
-> ```
-
 ## Configuration
 
-Consult the [upstream
+Consult the relevant manual pages and the [upstream
 documentation](https://iwd.wiki.kernel.org/networkconfigurationsettings) for
 options and examples available to configure the daemon and networks as described
 below.
 
 ### Daemon configuration
 
 The main configuration file is located in `/etc/iwd/main.conf`. If it does not
-exist, you may create it.
+exist, you may create it. It is documented in `iwd.config(5)`.
 
 ### Network configuration
 
-IWD stores information on known networks, and reads information on
-pre-provisioned networks from network configuration files located in
-`/var/lib/iwd`; IWD monitors the directory for changes. Network configuration
-filenames consist of the encoding of the SSID followed by `.open`, `.psk`, or
-`.8021x` as determined by the security type.
+Network configuration is documented in `iwd.network(5)`, including examples. IWD
+stores information on known networks, and reads information on pre-provisioned
+networks from network configuration files located in `/var/lib/iwd`; IWD
+monitors the directory for changes. Network configuration filenames consist of
+the encoding of the SSID followed by `.open`, `.psk`, or `.8021x` as determined
+by the security type.
 
 A simple example of a WPA2/PSK secured network configuration at
-`/var/lib/iwd/<ssid>.psk` contains the pre-shared key, and optionally the
-plaintext password.
+`/var/lib/iwd/<ssid>.psk` contains the plaintext password.
 
 ```
 [Security]
-PreSharedKey=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 Passphrase=<password>
 ```
+
+## Troubleshooting
+
+IWD can attach to your wireless interface (e.g. `wlan0`) before `udev` has the
+chance to rename it according to the consistent naming rules. If the device is
+already in use, `udev`'s renaming will fail, and there will be a `udevd` error,
+which might be printed to your screen, similar to this one:
+
+```
+[   39.441723] udevd[1100]: Error changing net interface name wlan0 to wlp59s0: Device or resource busy
+[   39.442472] udevd[1100]: could not rename interface '3' from 'wlan0' to 'wlp59s0': Device or resource busy
+```
+
+To solve this error, you can either disable consistent naming, by adding
+`net.ifnames=0` to your kernel [cmdline](../kernel.md#cmdline) or creating a
+symbolic link from `dev/null` to `/etc/udev/rules.d/80-net-name-slot.rules`,
+therefore masking the renaming rule; or configure the `iwd` service so it
+ignores interfaces with inconsistent names, which can be done by adding the
```

Comment:
in `/etc/iwd/main.conf`, UseDefaultInterface=true

Reply:
`UseDefaultInterface=True` works for me as well. In light of this, and to avoid confusion expressed by @flexibeast, I recomment that the whole section read:
<pre>
## Troubleshooting

By default, IWD will create and destroy the wireless interfaces (e.g., `wlan0`) that it manages. This can interfere with `udevd`, which may attempt to rename the interface using its rules for persistent network interface names. The following messages may be printed to your screen as a symptom of this interference:
```
[   39.441723] udevd[1100]: Error changing net interface name wlan0 to wlp59s0: Device or resource busy
[   39.442472] udevd[1100]: could not rename interface '3' from 'wlan0' to 'wlp59s0': Device or resource busy
```
A simple fix is to prevent IWD from manipulating the network interfaces in this way by adding `UseDefaultInterface=true` to the `[General]` section of `/etc/iwd/main.conf`

An alternative approach is to disable the use of persistent network interface names by `udevd`. This may be accomplished either by adding `net.ifnames=0` to your kernel [cmdline](../kernel.md#cmdline) or by creating a symbolic link to `/dev/null` at `/etc/udev/rules.d/80-net-name-slot.rules` to mask the renaming rule. The alternative approach will affect the naming of all network devices.
</pre>

Reply:
Definitely better to use an option in the native config file! Thanks :)

Reply:
Thanks a lot, I've updated it!